### ISSUE TITLE
Update docs for 0.15.0

### ISSUE THREAD

**Issue by Technologicat** [Technologicat/unpythonic#75]
- Check the changelog for what still needs updating; it's kept up to date during development.
- Update the *This document is up-to-date for vx.x.x* tags.

**Comment by Technologicat** [comment]
This should be pretty much done as of cef82ca, but I'm keeping this open for a couple more days in case I think of anything I've missed.

**Comment by Technologicat** [comment]
A-ha! The macro xmas tree needs to be better documented.

Right now we say in which order the syntax transformations should run, but that doesn't mean it's the correct ordering for the macros, since some (but not all) of them expand inside-out. The documentation mentions this caveat, but it would be nice to have a second xmas tree that gives the actual ordering they should be put into a `with ...` header.

Until this is fixed, there are a few valid combo examples in the dialects, and some more can be hunted down from the macro unit tests. The general method to determine the ordering is noted in the docs, but involves looking at source code.

**Comment by Technologicat** [comment]
Add to readings:

- https://legacy.cs.indiana.edu/~dyb/pubs/monadicDC.pdf
- https://github.com/manuel/wat-js
- https://github.com/piokuc/pywat

**Comment by Technologicat** [comment]
As of e9f0243, the xmas macro combo doc is done.

**Comment by Technologicat** [comment]
This is a big documentation update, still ongoing; as of 5b0b61b, about 70% of `features.md` has been reviewed and updated.

Once that file is done, the major remaining effort is to update `macros.md`.

**Comment by Technologicat** [comment]
As of 54e3fce, at 87% of `features.md`.

**Comment by Technologicat** [comment]
As of ad535ee, 100% of `features.md` done.

**Comment by Technologicat** [comment]
As of 78643c7, 20% of `macros.md` done.

At least the section on `let` constructs didn't need that many updates, after I'd already updated it once after upgrading the code.

**Comment by Technologicat** [comment]
As of e0ccd1c, 92% of `macros.md` done. This will be finished soon!

**Comment by Technologicat** [comment]
As of 2c30a69, 100% of `macros,md` done.

**Comment by Technologicat** [comment]
As of 5c5b682, *coming soon* notice removed from README. This is finally done!
